kuma/static/js/payments-confirmation.js

```diff
@@ -0,0 +1,33 @@
+(function(win) {
+    'use strict';
+
+    // Fire analytic events only if the user has followed payment
+    // flow denoted by `amountSubmitted` session varible.
+    var amountSubmittedStoreKey = 'submissionDetails';
+    var submissionDetails = JSON.parse(sessionStorage.getItem(amountSubmittedStoreKey));
+
+    // Default to no data
+    var amountSubmitted = submissionDetails.amount || 0;
+    var submissionPage = submissionDetails.page || '';
+
+    var path = win.location.pathname;
+    if (path.includes('/payments/success') && submissionDetails) {
```

Comment:
Both the conditional statements depend on its conditions evaluating to true. Could there ever be a scenario then where the `submissionDetails` would be stored in session storage but not contain any data? i.e. line 7 above could return an empty `Object`?

I am asking because if the item is not found in `sessionStorage`, then `submissionDetails` would be null and neither of the conditionals will pass and execute. This means there is no real need for lines 9-11

@joshJarr Thoughts?

Reply:
That's a fair point, I guess we're never really expecting a blank object, but if for some reason there is I was using the `||` to ensure `null` isn't passed to the events to prevent a runtime error or the wrong event parameters being sent.

Reply:
Thanks @schalkneethling, I was just about to ask if you could take a look.

I'm a belt and bracers dev, so I like code that works even if someone's session is screwed up.

Reply:
Actually, this will cause an error if there is no session storage since we'd be trying to access `null.amount`.
Looks like I should check `submissionDetails` and ensure its truthy before line 10 & 11.